homeassistant/components/minecraft_server/__init__.py

```diff
@@ -176,17 +171,21 @@ async def async_update(self, now: datetime | None = None) -> None:
     async def _async_status_request(self) -> None:
         """Request server status and update properties."""
         try:
-            status_response = await self._hass.async_add_executor_job(
-                self._mc_status.status, self._MAX_RETRIES_STATUS
-            )
+            status_response = await self._server.async_status()
 
             # Got answer to request, update properties.
             self.version = status_response.version.name
             self.protocol_version = status_response.version.protocol
             self.players_online = status_response.players.online
             self.players_max = status_response.players.max
             self.latency_time = status_response.latency
-            self.motd = (status_response.description).get("text")
+
+            # For some servers MOTD is not the key 'text' but rather the description directly.
+            try:
+                self.motd = (status_response.description).get("text")  # type: ignore[attr-defined]
+            except AttributeError:
+                self.motd = status_response.description
+
```

Comment:
```suggestion
            if isinstance(status_response.description, dict):
                self.motd = status_response.description.get("text")  # type: ignore[attr-defined]
            else:
                self.motd = status_response.description

```

Reply:
Thanks for looking into my changes. Your proposal would be better, agreed. 
But in the mean time I bumped mcstatus to version 11.0.0-rc3, which fixes a dependency problem with dnspython within HA. In this new version there is a new MOTD parser, which I am using now. This (hopefully) doesn't require the special handling above anymore.

`self.motd = status_response.motd.to_plain()`